backend/src/apiserver/webhook/pipelineversion_webhook.go

```diff
@@ -134,13 +141,23 @@ func (p *PipelineVersionsWebhook) Default(ctx context.Context, obj runtime.Objec
 	pipeline := &k8sapi.Pipeline{}
 	nsName := types.NamespacedName{Namespace: pipelineVersion.Namespace, Name: pipelineVersion.Spec.PipelineName}
 
-	err := p.Client.Get(ctx, nsName, pipeline)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			return newBadRequestError("The spec.pipelineName doesn't map to an existing Pipeline object")
+	// Because controller-client cache cannot be up to date when retrieving the pipeline,
+	// this loop will retry 10 times and wait 500ms in each retry.
+	for tries := 0; tries < maxRetries; tries++ {
+		err := p.Client.Get(ctx, nsName, pipeline)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				if tries < maxRetries {
```

Comment:
if tries < maxRetries this condition looks like it's always true

Reply:
Good catch! I fixed that.